main/common/plugins.js

```diff
@@ -6,27 +8,70 @@ const execa = require('execa');
 const makeDir = require('make-dir');
 const {ipcMain: ipc} = require('electron-better-ipc');
 const packageJson = require('package-json');
-const {satisfies} = require('semver');
 
 const {app, Notification} = electron;
 
-const Plugin = require('../plugin');
+const {refreshRecordPluginItems} = require('../menus');
 const {openConfigWindow} = require('../config');
 const {openPrefsWindow} = require('../preferences');
 const {notify} = require('./notifications');
 const {track} = require('./analytics');
+const {InstalledPlugin, NpmPlugin, recordPluginServiceState} = require('../plugin');
+const {showError} = require('../utils/errors');
 
 class Plugins {
   constructor() {
     this.yarnBin = path.join(__dirname, '../../node_modules/yarn/bin/yarn.js');
     this._makePluginsDir();
     this.appVersion = app.getVersion();
+    this.refreshRecordPluginServices();
   }
 
   setUpdateExportOptions(updateExportOptions) {
     this.updateExportOptions = updateExportOptions;
   }
 
+  async enableService(service) {
+    const wasEnabled = recordPluginServiceState.get(service.title) || false;
+
+    if (wasEnabled) {
+      recordPluginServiceState.set(service.title, false);
+      return this.refreshRecordPluginServices();
+    }
+
+    if (service.willEnable) {
+      try {
+        const canEnable = await service.willEnable();
+
+        if (canEnable) {
+          recordPluginServiceState.set(service.title, true);
+        }
+      } catch (error) {
+        showError(error, {title: `Something went wrong while enabling “${service.title}”`});
```

Comment:
Can you make a `showErrorAndReport()` utility method that shows the error and also reports to sentry?

Reply:
```suggestion
        showError(error, {title: `Something went wrong while enabling “${service.title}”`, reportToSentry: true});
```